public/app/core/utils/UserProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+
+export interface Props {
+  children: (props: any) => React.ReactNode;
```

Comment:
children should be more specific about the exposed API. Ideally, the children type could be:

`children: (api: UserAPI, user?: User) => JSX.Element`

where `UserAPI` interface specifies what can be performed on user (change password, refresh user, etc), and `user` is a object representing user that was fetched.

Reply:
Okay. Makes sense. So I would make a type for the API then? I don't have user yet because that's not needed at this point.

Reply:
That's cool. Yes, you need an API interface